src/control/cmd/daos_server/network.go

```diff
@@ -89,10 +98,34 @@ type networkListCmd struct {
 	logCmd
 }
 
-// List the supported providers
+// List the available providers
```

Comment:
So these two functions are pretty much identical except for the provider selection code and the call to `pretty.PrintHostFabricMap()` toggling `onlyProviders`, right? Seems like it would be better to extract the common code to a helper function.

Reply:
Done.  I followed what you had done in dmg, which was to implement two separate handlers, one for the scan and one for the list.  But I agree that consolidating it is good.  That said, I'm not sure if it's helpful to have the list functionality at all.  I added it at the time thinking it might be helpful, but I'm not so sure it adds much utility any longer.  If you want it removed, this go around, let me know.  I also don't want to assume that you do want the dmg commands consolidated if we keep the list(), so I've left that as-is for now.

Reply:
Ah, OK. Well, now that the list of providers is printed above the tables in scan, it's probably best to just remove the list command.